cocos/core/components/ui-base/canvas.ts

```diff
@@ -368,21 +400,22 @@ export class Canvas extends Component {
     }
 
     protected _updateTargetTexture () {
-        if (!this._camera) {
-            return;
-        }
+        if (!this._cameraComponent) { return; }
+
+        const camera = this._cameraComponent;
+        if (!camera.camera) { return; }
 
-        const camera = this._camera;
         if (!this._targetTexture) {
-            camera.changeTargetWindow();
+            camera.camera.changeTargetWindow();
             camera.orthoHeight = game.canvas!.height / view.getScaleY() / 2;
```

Comment:
camera.camera is not clear, you can change the variable name

Reply:
sure